src/graphic/helper/dashStyle.ts

```diff
@@ -0,0 +1,30 @@
+import { isArray, isNumber, isString } from '../../core/util';
+
+function normalizeDashArray(lineType: any): number[] | false {
+    if (!lineType || lineType === 'solid'
+        || (isString(lineType) && lineType !== 'dashed' && lineType !== 'dotted')) {
+        return [];
+    }
+    if (isArray(lineType)) {
+        return lineType;
+    }
+    if (isNumber(lineType)) {
+        return [lineType];
+    }
+}
+
+export function normalizeLineDash(lineType: any, lineWidth?: number): number[] | false {
+    if (lineWidth != null && lineWidth <= 0) {
+        return [];
+    }
+    const dashArray = normalizeDashArray(lineType);
+    if (dashArray) {
+        return dashArray;
+    }
+    lineWidth = lineWidth || 1;
+    return lineType === 'dashed' 
+        ? [4 * lineWidth, 2 * lineWidth]
+        : lineType === 'dotted' 
+            ? [lineWidth]
+            : [];
+}
```

Comment:
Perhaps `normalizeDashArray` can be combined with this condition to simplify the check.

```ts
 return lineType === 'dashed' 
        ? [4 * lineWidth, 2 * lineWidth]
        : lineType === 'dotted' 
            ? [lineWidth]
            : isNumber(lineType)
                ? [lineType] : isArray(lineType) ? lineType : null;
```

Also, I think we should return `null` if `lineDash` is invalid to reduce the `setLineDash` call as more as possible.

Reply:
Thank you, I learned this from your guide and will be careful of it in the future.